serverSocket.js

```diff
@@ -26,16 +27,28 @@ function init(server) {
       return lastMessageB.sentAt - lastMessageA.sentAt;
     });
 
-    const chats = userChats;
+    let chats = userChats;
     // find all the users in each chat ad add it as a property 'members'
-    return await Promise.all(
+    chats = await Promise.all(
       chats.map(async (chat) => {
         const members = await UserChat.find({ chat: chat._id })
           .populate('user')
           .exec();
         return { ...chat._doc, members: members.map((member) => member.user) };
       })
     );
+
+    chats = chats.map((chat) => {
+      chat.members = chat.members.map((member) => {
+        return {
+          ...member._doc,
+          online: onlineUsers.has(member._id.toString()),
+        };
+      });
+      return chat;
+    });
+
+    return chats;
```

Comment:
Is this possible to make a chain of promises instead to assign to chats 2 times and make it const?

Reply:
Thanks nico!